tests_e2e/tests/agent_status/agent_status.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+import datetime
+from time import sleep
+
+from assertpy import assert_that
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Validates the agent status is updated without any other goal state changes
+#
+
+from azure.mgmt.compute.models import VirtualMachineInstanceView
+
+from tests_e2e.tests.lib.agent_test import AgentTest
+from tests_e2e.tests.lib.agent_test_context import AgentTestContext
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine_client import VirtualMachineClient
+
+
+def validate_instance_view_vmagent_status(instance_view: VirtualMachineInstanceView):
```

Comment:
why use global functions? shouldn't these be methods of the test class?

Reply:
pycharm suggested making them functions since they were static

Reply:
i think they should be methods of the test class though, I'll move them in the test class